lib/irb.rb

```diff
@@ -640,7 +646,12 @@ def handle_exception(exc)
           message = exc.full_message(order: :top)
           order = :top
         end
-        message = convert_invalid_byte_sequence(message)
+        message = convert_invalid_byte_sequence(message, exc.message.encoding)
+        message = encode_with_invalid_byte_sequence(message, IRB.conf[:LC_MESSAGES].encoding) if message.encoding != IRB.conf[:LC_MESSAGES].encoding
+        #message = message.encode(IRB.conf[:LC_MESSAGES].encoding).scrub { |c|
+        #  puts "!!!!!!!!!!! #{c.inspect} is a scrub in #{IRB.conf[:LC_MESSAGES].encoding}"
+        #  c.bytes.map{ |b| "\\x#{b.to_s(16).upcase}" }.join
+        #}
```

Comment:
Perhaps these lines were sketching not meant for production use?

Reply:
Good catch, thank you so much!